### PR TITLE
Update repositories.txt for ams-OSRAM[-Group]

### PR DESCRIPTION
**Pull request by maarten-pennings**
These libraries are written by ams-OSRAM for their products. 

The GitHub page `https://github.com/ams-OSRAM` used to be claimed by an unknown party, therefore ams-OSRAM was using `https://github.com/ams-OSRAM-Group` (note the suffix `-Group`). After some trademark claims, ams-OSRAM was given ownership of `https://github.com/ams-OSRAM` and moved their repos.

This commit updates the Arduino registrations to that new location.

### CODE REVIEW

**Review by per1234 — COMMENTED** [review]
Hello @maarten-pennings. I am confirming your request.

Unlike submissions, this type of request is still handled manually. The Library Manager index maintainer occasionally comes through to clear all the outstanding requests. They will merge this pull request when it is done. You will get a notification from GitHub at that time.

After that, you can monitor the indexing of the library via its dedicated logs web page.
Information about that here:
https://github.com/arduino/library-registry/blob/main/FAQ.md#can-i-check-on-library-releases-being-added-to-library-manager

**Review by MatteoPologruto — APPROVED** [review]
Done:
```
2024/10/10 14:54:06 Loaded 7357 libraries from DB
Changing URL of library OSP 2wireSPI aospi from https://github.com/ams-OSRAM-Group/OSP_aospi.git to https://github.com/ams-OSRAM/OSP_aospi.git
Success!
```
```
2024/10/10 14:58:16 Loaded 7357 libraries from DB
Changing URL of library OSP ResultCodes aoresult from https://github.com/ams-OSRAM-Group/OSP_aoresult.git to https://github.com/ams-OSRAM/OSP_aoresult.git
Success!
```
```
2024/10/10 15:00:00 Loaded 7357 libraries from DB
Changing URL of library OSP Telegrams aoosp from https://github.com/ams-OSRAM-Group/OSP_aoosp.git to https://github.com/ams-OSRAM/OSP_aoosp.git
Success!
```
```
2024/10/10 15:01:19 Loaded 7357 libraries from DB
Changing URL of library OSP CommandInterpreter aocmd from https://github.com/ams-OSRAM-Group/OSP_aocmd.git to https://github.com/ams-OSRAM/OSP_aocmd.git
Success!
```
```
2024/10/10 15:02:57 Loaded 7357 libraries from DB
Changing URL of library OSP Middleware aomw from https://github.com/ams-OSRAM-Group/OSP_aomw.git to https://github.com/ams-OSRAM/OSP_aomw.git
Success!
```
```
2024/10/10 15:06:39 Loaded 7357 libraries from DB
Changing URL of library OSP UIDriversOSP32 aoui32 from https://github.com/ams-OSRAM-Group/OSP_aoui32.git to https://github.com/ams-OSRAM/OSP_aoui32.git
Success!
```

**Comment by maarten-pennings** [comment]
Thanks for the work @MatteoPologruto and @per1234

**Pull request merged**
